test/Microsoft.Framework.DependencyInjection.Tests/ServiceProviderExtensionsTest.cs

```diff
@@ -70,73 +105,83 @@ public void NonGeneric_GetRequiredServices_Returns_SingleService()
+            var services = serviceProvider.GetServices(typeof(IBar));
 
             // Assert
             foreach (var service in services)
             {
-                Assert.IsAssignableFrom<IAnotherFoo>(service);
+                Assert.IsAssignableFrom<IBar>(service);
             }
             Assert.Equal(2, services.Count());
         }
 
         [Fact]
-        public void Throws_WhenNo_Services_AreRegistered()
+        public void GetServices_Returns_EmptyArray_WhenNoServicesAvailable()
         {
             // Arrange
             var serviceProvider = CreateTestServiceProvider(0);
 
-            // Act + Assert
-            ExceptionAssert.Throws<InvalidOperationException>(() => serviceProvider.GetRequiredServices<IFoo>(),
-                $"No service for type '{typeof(IFoo)}' has been registered.");
+            // Act
+            var services = serviceProvider.GetServices<IFoo>();
+
+            // Assert
+            Assert.Empty(services);
+            Assert.IsType<IFoo[]>(services);
         }
 
         [Fact]
-        public void Throws_WhenNo_Services_AreRegistered_NonGeneric()
+        public void NonGeneric_GetServices_Returns_EmptyArray_WhenNoServicesAvailable()
         {
             // Arrange
             var serviceProvider = CreateTestServiceProvider(0);
 
-            // Act + Assert
-            ExceptionAssert.Throws<InvalidOperationException>(() => serviceProvider.GetRequiredServices(typeof(IFoo)),
-                $"No service for type '{typeof(IFoo)}' has been registered.");
+            // Act
+            var services = serviceProvider.GetServices(typeof(IFoo));
+
+            // Assert
+            Assert.Empty(services);
+            Assert.IsType<IFoo[]>(services);
         }
 
         [Fact]
-        public void Throws_WhenNo_Services_AreAvailable()
+        public void GetServices_WithBuildServiceProvider_Returns_EmptyList_WhenNoServicesAvailable()
         {
             // Arrange
             var serviceCollection = new ServiceCollection();
             serviceCollection.AddInstance<IEnumerable<IFoo>>(new List<IFoo>());
-
             var serviceProvider = serviceCollection.BuildServiceProvider();
 
-            // Act + Assert
-            ExceptionAssert.Throws<InvalidOperationException>(() => serviceProvider.GetRequiredServices<IFoo>(),
-                $"No service for type '{typeof(IFoo)}' has been registered.");
+            // Act
+            var services = serviceProvider.GetServices<IFoo>();
+
+            // Assert
+            Assert.Empty(services);
+            Assert.IsType<List<IFoo>>(services);
         }
 
         [Fact]
-        public void Throws_WhenNo_Services_AreAvailable_NonGeneric()
+        public void NonGeneric_GetServices_WithBuildServiceProvider_Returns_EmptyList_WhenNoServicesAvailable()
         {
             // Arrange
             var serviceCollection = new ServiceCollection();
```

Comment:
Could we add a variant of this test (and the one above) where `List<Foo>` is not registered? Would be a better verification for the scenario where no services of a type are registered and we invoke `GetServices` on it.